### PR TITLE
Add button to add another instance type

### LINKED ISSUE #10049: Add ability to add another instance in contact modal
on `/openstack/pricing-calculator` ther `Get cost estimates` pop up modal gives you the option to add one instance type. There should be the option to add multiple instance types

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Added an option to add another instance type (up to 4) on pricing calculator contact form

## QA

- Visit page at: https://ubuntu-com-10054.demos.haus/openstack/pricing-calculator
- Click on the "Get cost estimates" button
- Click the "Add another instance" button
- Check Instance Type 2 appears. Then 3, then 4, then the button to add another instance should be disabled

## Issue / Card

Fixes #10049 

## Screenshots

Screenshot 2021-07-20 at 15 19 08

### CODE REVIEW

**Review by anthonydillon — CHANGES_REQUESTED** [review]
When jumping to the end without changing anything in the form or adding an instance I see three instances in the comment:
```
vCPUs
2

RAM [GB]
8

Ephemeral storage [GB]
8

Persistent storage [GB]
80

Number of instances
1000

vCPUs


RAM [GB]


Ephemeral storage [GB]


Persistent storage [GB]


Number of instances


vCPUs


RAM [GB]


Ephemeral storage [GB]


Persistent storage [GB]


Number of instances


vCPUs


RAM [GB]


Ephemeral storage [GB]


Persistent storage [GB]


Number of instances


Support level
fully-managed

Operations team size [FTEs]
12

Operations team average annual staff salary [USD]
125000

 CPU overcommitment ratio
2

RAM overcommitment ratio
1

Storage replication factor
3

Desired cloud utilisation ratio [%]
75

Cloud architecture
reference

Hardware renewal period [years]
3

Annual per-node hosting rent and electricity cost [USD]
4500

Annual per-node hardware installation and maintenance cost [USD]
1000

External network bandwidth
125000

Annual per-Gbps hosting external network cost
6000
```

**Comment by bethcollins92** [comment]
@anthonydillon I've removed the extra input titles if the instance isn't added, there are still `/n` added giving some new lines not needed, is this an issue?

**Comment by anthonydillon** [comment]
I think you can remove the `/n` as it results in a large gap in the result if only one instance is posted.

File: templates/openstack/pricing-calculator.html [hunk 1]
```diff
@@ -129,6 +128,34 @@ <h3 class="p-heading--4">Still not sure which cloud platform suits you best?</h3
...
+        document.querySelectorAll("#instance-type")
+      );
+      const length = instanceTypes.length;
+      for (let i = 0; i < length; i++) {
+        if (instanceTypes[i].classList.contains("u-hide")) {
+          instanceTypes[i].classList.remove("u-hide");
+          //when new instance is added add p-heading--five class to h3s for inputs only to capture values in comment.
+          const inputTitles = instanceTypes[i].querySelectorAll("h3")
+          inputTitles.forEach(inputTitle => {
+            if(!inputTitle.classList.contains("p-heading--four")) {
```
Comment (anthonydillon) [inline templates/openstack/pricing-calculator.html:145]:
```suggestion
            if (!inputTitle.classList.contains("p-heading--four")) {
```

Comment (anthonydillon) [reply templates/openstack/pricing-calculator.html:145]: Interesting hack.

File: templates/openstack/pricing-calculator.html [hunk 2]
```diff
@@ -129,6 +128,34 @@ <h3 class="p-heading--4">Still not sure which cloud platform suits you best?</h3
...
+              inputTitle.classList.add("p-heading--five")
```
Comment (anthonydillon) [inline templates/openstack/pricing-calculator.html:146]: If you add a class that exists it does not duplicate the class. So you don't need the conditional above.

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Code is looking good, a few comments in the code.

File: templates/openstack/pricing-calculator.html [hunk 3]
```diff
@@ -129,6 +128,34 @@ <h3 class="p-heading--4">Still not sure which cloud platform suits you best?</h3
 };
 updateFormFields();
 
+const addAnotherInstance = () => {
+  document.addEventListener("contactModalLoaded", (event) => {
+    const button = document.querySelector("#add-another-instance");
+    button.addEventListener("click", () => {
+      const instanceTypes = Array.from(
+        document.querySelectorAll("#instance-type")
```
[the remark this answers is not in the document]
Comment (bethcollins92) [reply templates/openstack/pricing-calculator.html]: ahh yes, done!

File: templates/openstack/pricing-calculator.html (hunk 1 above)
Comment (bethcollins92) [reply templates/openstack/pricing-calculator.html]: I actually updated the input labels to be a h4 as this makes more sense semantically, so don't need to check if it's the title or the input label

**Comment by bethcollins92** [comment]
> I think you can remove the `/n` as it results in a large gap in the result if only one instance is posted.

Sure, removed them

File: templates/shared/forms/interactive/openstack-cost-calculator.html [hunk 4]
```diff
@@ -240,3 +338,5 @@ <h3 class="p-heading--two">Thank you!</h3>
     </div>
   </div>
 </div>
+
+<script>console.log("hello")</script>
```
Comment (anthonydillon) [inline templates/shared/forms/interactive/openstack-cost-calculator.html:342]: Hello :wave:

**Review by anthonydillon — APPROVED** [review]
Other than the small comment in the code, good to go :+1:

File: templates/shared/forms/interactive/openstack-cost-calculator.html (hunk 4 above)
Comment (bethcollins92) [reply templates/shared/forms/interactive/openstack-cost-calculator.html]: 👋👋👋 wups!

**Pull request merged**